MapboxCoreNavigation/CLLocation.swift

```diff
@@ -191,4 +190,12 @@ extension CLLocation {
         }
         return true
     }
+    
+    func shifted(to newTimestamp: Date) -> CLLocation {
+        return CLLocation(coordinate: coordinate, altitude: altitude, horizontalAccuracy: horizontalAccuracy, verticalAccuracy: verticalAccuracy, course: course, speed: speed, timestamp: newTimestamp)
```

Comment:
It occurs to me that this method is lossy if `CLLocation.floor` happens to be set, but I don’t think that poses a practical problem at the moment.

Apple could add new properties to CLLocation in the future, which make this method even lossier. To guard against that, we could have a unit test that shifts a CLLocation to its own timestamp, then checks whether the original and shifted objects are equal.

Reply:
CLLocation doesn't implement equatable but I added a test for shifting and FixLocation->CLLocation.